pkg/operator/resources/resources.go

```diff
@@ -170,6 +169,83 @@ func UpdateAPI(apiConfig *userconfig.API, models []spec.CuratedModelResource, pr
 	return nil, msg, err
 }
 
+func Patch(configBytes []byte, configFileName string, force bool) ([]schema.DeployResult, error) {
+	apiConfigs, err := spec.ExtractAPIConfigs(configBytes, types.AWSProviderType, configFileName, &config.Cluster.Config)
+	if err != nil {
+		return nil, err
+	}
+
+	results := make([]schema.DeployResult, 0, len(apiConfigs))
+	for i := range apiConfigs {
+		apiConfig := &apiConfigs[i]
+		result := schema.DeployResult{}
+
+		apiSpec, msg, err := patchAPI(apiConfig, configFileName, force)
+		if err == nil && apiSpec != nil {
+			apiEndpoint, _ := operator.APIEndpoint(apiSpec)
+
+			result.API = &schema.APIResponse{
+				Spec:     *apiSpec,
+				Endpoint: apiEndpoint,
+			}
+		}
+
+		result.Message = msg
+		if err != nil {
+			result.Error = errors.ErrorStr(err)
+		}
+
+		results = append(results, result)
+	}
+	return results, nil
+}
+
+func patchAPI(apiConfig *userconfig.API, configFileName string, force bool) (*spec.API, string, error) {
+	deployedResource, err := GetDeployedResourceByName(apiConfig.Name)
+	if err != nil {
+		return nil, "", err
+	}
+
+	if deployedResource.Kind == userconfig.TrafficSplitterKind {
+		return trafficsplitter.UpdateAPI(apiConfig, force)
+	} else if deployedResource.Kind != userconfig.BatchAPIKind && deployedResource.Kind != userconfig.RealtimeAPIKind {
+		return nil, "", ErrorOperationIsOnlySupportedForKind(*deployedResource, userconfig.RealtimeAPIKind, userconfig.BatchAPIKind, userconfig.TrafficSplitterKind) // unexpected
+	}
+
+	prevAPISpec, err := operator.DownloadAPISpec(deployedResource.Name, deployedResource.ID())
+	if err != nil {
+		return nil, "", err
+	}
+
+	bytes, err := config.AWS.ReadBytesFromS3(config.Cluster.Bucket, prevAPISpec.ProjectKey)
+	if err != nil {
+		return nil, "", err
+	}
+
+	projectFileMap, err := archive.UnzipMemToMem(bytes)
+	if err != nil {
+		return nil, "", err
+	}
+
+	projectFiles := ProjectFiles{
+		ProjectByteMap: projectFileMap,
+	}
+
+	models := []spec.CuratedModelResource{}
+	err = ValidateClusterAPIs([]userconfig.API{*apiConfig}, &models, projectFiles)
+	if err != nil {
+		err = errors.Append(err, fmt.Sprintf("\n\napi configuration schema can be found here:\n  → Realtime API: https://docs.cortex.dev/v/%s/deployments/realtime-api/api-configuration\n  → Batch API: https://docs.cortex.dev/v/%s/deployments/batch-api/api-configuration\n  → Traffic Splitter: https://docs.cortex.dev/v/%s/deployments/realtime-api/traffic-splitter", consts.CortexVersionMinor, consts.CortexVersionMinor, consts.CortexVersionMinor))
+		return nil, "", err
+	}
+
+	switch deployedResource.Kind {
+	case userconfig.RealtimeAPIKind:
+		return realtimeapi.UpdateAPI(apiConfig, models, prevAPISpec.ProjectID, force)
```

Comment:
I'm pretty sure that it's the case, but just wanted to confirm that project ID does not include the API config file